### ISSUE TITLE
You can't paste number with Three Decimal Digits (or more ) in IOU page - reported by @ahmdshrif

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#7911]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Copy number with 3 dismissal digits e.g. 55.333
2. Navigate to IOU request 
3. Paste the value copied

## Expected Result:
User should be able to paste the number by keeping 2 decimals 

## Actual Result:
Value is not pasted at all 

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.40-0
**Reproducible in staging?:**  Y 
**Reproducible in production?:**  Y 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 

https://user-images.githubusercontent.com/43995119/155665063-1a675698-3309-484d-8b2c-e179e27d45bc.mp4

**Expensify/Expensify Issue URL:** 
**Issue reported by:**  @ahmdshrif
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1644795303142109

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @ctkochan22 (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by ctkochan22** [comment]
This seems _super_ edge case. If the user types in the number, I assume it stops at 5.33. But I suppose its worth fixing? Downgrading to monthly and looking when I have time

**Comment by ctkochan22** [comment]
Interested. Will look later this month

**Comment by ctkochan22** [comment]
Punting

**Comment by ctkochan22** [comment]
@mvtglobally If someone reports this again, please label as not an issue.

**Comment by ahmdshrif** [comment]
Hi  @ctkochan22  do i will be eligible for report bonus here ? 

Can you please add external lable

**Comment by ctkochan22** [comment]
I don't think so? Because its not a bug, but let me ask.

**Comment by ahmdshrif** [comment]
Thanks @ctkochan22 for response.  

But As i remember Roray say it's BUG we should fix when i report this in slack.  
and  then you decide we shouldn't do any thing Because it's minor case. And based on some logic we have in code . (For me still see it bug at least we should show  messas to user to let he  know what is the error. but i agree withe you  it's minor case ) 

That's why i think we should add external lable to discuss this case with the CM.

**Comment by ctkochan22** [comment]
Sure we can discuss.

**Comment by ctkochan22** [comment]
The issue here is that whenever a user inputs something that isn't a number in this case, we don't update the state. When a user types in `1`, we validate that input then update the `state.amount` to `1`. When the user then types in `a`, we check the character, identify that its not a number, then we do _not_ update the `state.amount` to `1a`. We set the state to the previous value `1`.

**I don't think this is a bug.** My reasoning is thinking about the case where a user accidentally has a string in their clipboard. So they try and paste the value `asdf123@gmail.com` into the amount field. I think the expected result is to not past any string values into the field. 

In the rare case that a user copy/pastes a whole amount. In this case, three digits instead of two `22.123`, we get the input. See that its an invalid amount. Then we update the state with the previous value, and the previous value is empty. I think it works as it should. I don't think we should arbitrarily cut the value to two decimal points. Nor do I think we should round up or down.

**Comment by ctkochan22** [comment]
@mallenexpensify let me know what you think

**Comment by mallenexpensify** [comment]
Weird, I thought I commented. I def read (then re-read to try to make sure I understood everything). 
I'm on board with your reasoning @ctkochan22 , I hadn't even tried to paste numbers or letters into the amount field before.  

@ahmdshrif can you test to see how it works for a currency with more than 2 decimal places? I wasn't able to update my default currency (possibly because I'm on the Expensify policy?).  Here's a list of decimal places by currency https://docs.adyen.com/development-resources/currency-codes

**Comment by ahmdshrif** [comment]
@mallenexpensify Same thing with the currency with 3 decimal place . 
 https://user-images.githubusercontent.com/21364901/172261767-f28308fe-00db-4981-b678-d31d490fb804.mp4

**Comment by mallenexpensify** [comment]
Thanks for the vid @ahmdshrif, what I was wondering was, if it's a currency with three decimal places, that a user would be able to paste in a number with 3 decimal places.  I'm assuming, in your vid, that you were trying to paste in a number with more than three, is that correct?

For payments for bugs and feature requests, for the most part, like this issue, code has to be deployed to production in order to be compensated. 
https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#:~:text=Payment%20will%20be%20made%207%20days%20after%20code%20is%20deployed%20to%20production%20if

**Comment by ahmdshrif** [comment]
>Thanks for the vid @ahmdshrif, what I was wondering was, if it's a currency with three decimal places, that a user would be able to paste in a number with 3 decimal places. I'm assuming, in your vid, that you were trying to paste in a number with more than three, is that correct?


@mallenexpensify  no, I try to paste a number with 3 decimal places `1.333`   and the currency is  KDW. 
the current code only accepts 2 numbers with 2 decimal places when you enter in by keyboard or past it. 

>For payments for bugs and feature requests, for the most part, like this issue, code has to be deployed to production in order to be compensated.

ok, I just asking about it.

**Comment by mallenexpensify** [comment]
ah ha! Then that is a bug.  If a currency accepts more than two decimal places then the user should be able to paste in, up to, the max number of decimal places for each type of currency.   

@ctkochan22 I think we should fix/update this, do you think it can be worked on externally? If so someone will first have to do an audit of the different currencies.

**Comment by ctkochan22** [comment]
I can do this this week

**Comment by ctkochan22** [comment]
Hmm we don't even allow triple digits on oldDot for these currencies.

But I don't think we can do this externally, we should add this as a constant in /expensify-common

**Comment by mallenexpensify** [comment]
If we can't do this Externally, will you add the `Internal` label? I'll stay assigned to issue payment to @ahmdshrif if the issue ever gets fixed.

**Comment by ctkochan22** [comment]
I don't think this is super pressing. I am hoping to do this still

**Comment by mallenexpensify** [comment]
@adelekennedy assigned you as CM to issue payment to @ahmdshrif if we fix this internally

**Comment by ctkochan22** [comment]
Lets bump this to a monthly. Unless we are actually seeing a lot of customers dealing with this issue

**Comment by ctkochan22** [comment]
not a priority atm

**Comment by ctkochan22** [comment]
Slapping a not a priority. Let me know if you all think otherwise.

**Comment by mallenexpensify** [comment]
@ctkochan22 wanna unassign and open up to the pool?  I think we should consider fixing this at some point, although it's not a priority

**Comment by adelekennedy** [comment]
Still pending on fixing this but will remit payment if we do!

**Comment by adelekennedy** [comment]
@ctkochan22 what do you think of Matt's comment [here](https://github.com/Expensify/App/issues/7911#issuecomment-1230948353)?

**Comment by ctkochan22** [comment]
Sure, i'll unassign. Sorry about that

**Comment by ctkochan22** [comment]
Who ever works on this, you'll have to figure out where we grab the currency list and references to in expensify-common. And work with that there

**Comment by adelekennedy** [comment]
Where are we at here? We've opened this up to anyone who wants to work on it (with the caveat we'll pay out to @ahmdshrif if we fix internally) @mallenexpensify should we make this an external issue again?

**Comment by mallenexpensify** [comment]
https://github.com/Expensify/App/issues/7911#issuecomment-1149491041
It appears to need to be fixed internally, I'd leave it a monthly to see if anyone picks up. You should stay assigned @adelekennedy , in case it does get fix, @ahmdshrif will be due compensation if so

**Comment by ahmdshrif** [comment]
> But I don't think we can do this externally, we should add this as a constant in /expensify-common

@mallenexpensify I think @ctkochan22 thinks we can't fix it externally because of changes need in /expensify-common .

but I already contribute to this repo on 2 issues I get assigned for (as an external contributor ) 
https://github.com/Expensify/expensify-common/pull/436

if that's the only reason we can move it to external.

**Comment by mallenexpensify** [comment]
Thanks @ahmdshrif , asking internally for 👀

**Comment by adelekennedy** [comment]
@mallenexpensify Can you link the slack thread? Any progress here?

**Comment by mallenexpensify** [comment]
Blast from the past!  Sorry for leaving you hanging @adelekennedy .  I did check [here](https://expensify.slack.com/archives/C02NK2DQWUX/p1664472772300189), I'm going to close this because we currently don't support more than 2 decimals for currencies and it hasn't been a problem so far.  Maybe we'll address sometime down the line
